### PR TITLE
Development

### PR DESCRIPTION
**Pull request by rivalez**

### CODE REVIEW

File: battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java [hunk 1]
```diff
@@ -27,11 +27,12 @@
 
     @Override
     public void run() throws IOException {
-        String fromServer;
-        while (!("Provide ships").equals(fromServer = in.readLine())) {
+        String fromServer = in.readLine();
+        while (!("Provide ships").equals(fromServer)) {
```
Comment (brrt2) [inline battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java]: String is a bad idea here

File: battleship-client/src/main/java/com/academy/solid/nie/client/communication/Server.java [hunk 2]
```diff
@@ -16,6 +16,7 @@
      * Returns true if connection was successful.
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/communication/Server.java]: Return line as a description? Please make it a `@return` one instead.

File: battleship-client/src/main/java/com/academy/solid/nie/client/communication/Server.java [hunk 3]
```diff
@@ -43,6 +44,7 @@
 
     /**
      * Receive ships.
+     *
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/communication/Server.java]: No need to have method name repeated in a Java doc.

File: battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java [hunk 4]
```diff
@@ -27,11 +27,12 @@
 
     @Override
     public void run() throws IOException {
-        String fromServer;
-        while (!("Provide ships").equals(fromServer = in.readLine())) {
+        String fromServer = in.readLine();
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java]: Just to make sure: which logic you are after here - the one where `readLine` is called once or where each iteration calls it?

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Point2D.java [hunk 5]
```diff
@@ -23,15 +23,25 @@ public String toString() {
...
+     * @return information whether two points are the same
+     */
     @Override
     public boolean equals(Object o) {
-        if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
```
Comment (brrt2) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/Point2D.java]: getClass looks like a code smell...

File: battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java [hunk 6]
```diff
@@ -27,11 +27,12 @@
...
+        while (!("Provide ships").equals(fromServer)) {
             if (LOGGER.isLoggable(Level.INFO)) {
                 LOGGER.info("Server: " + fromServer);
             }
+            fromServer = in.readLine();
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java]: So, before the while, you are setting `fromServer` to whatever comes from server, in hopes in is `Provide ships`. As long as it is, you are changing it? What should `fromServer` have here, after this substitution is done?

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Board.java [hunk 7]
```diff
@@ -191,20 +192,20 @@ void makeMoves(List<Point2D> points) {
     }
 
     private void markSunkenShipOnPlayerBoard() {
-        allShips.stream().filter(ship -> isShipSunken(ship)).forEach(ship -> markShipAsSunken(ship));
+        allShips.stream().filter(this::isShipSunken).forEach(this::markShipAsSunken);
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/Board.java]: :+1:

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Board.java [hunk 8]
```diff
@@ -191,20 +192,20 @@ void makeMoves(List<Point2D> points) {
...
     }
 
     boolean isShipSunken(Ship ship) {
         return ship != null && !ship.isAlive();
     }
 
     void markShipAsSunken(Ship ship) {
-        getAllNeighborsOf(ship).stream().filter(e -> isInScope(e)).forEach(e -> boardFX.shoot(e));
+        getAllNeighborsOf(ship).stream().filter(this::isInScope).forEach(e -> boardFX.shoot(e));
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/Board.java]: :+1:

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Board.java [hunk 9]
```diff
@@ -191,20 +192,20 @@ void makeMoves(List<Point2D> points) {
...
     }
 
     private List<Point2D> getAllNeighborsOf(Ship ship) {
         return ship.getPositions().stream()
-                .map(e -> getAllNeighborsOf(e))
+                .map(this::getAllNeighborsOf)
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/Board.java]: :+1:

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java [hunk 10]
```diff
@@ -41,21 +43,21 @@
...
+        }
     }
 
-    boolean areAllShipsPlaced() {
-        return areAllShipsPlaced;
+    private Ship makeShip(Cell cell, Type type) {
+        return new Ship(producePoints(cell, type));
     }
 
     private Type shipOrientation(MouseEvent event) {
```
Comment (brrt2) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java]: empty method ?

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Cell.java [hunk 11]
```diff
@@ -39,24 +39,22 @@ boolean wasShot() {
...
         this.ship = newShip;
-        return this;
     }
 
     /**
      * Changes the cell color and marks it as a hit.
      * If there was a piece of a ship in the cell, it would return true.
      *
-     * @return
+     * @return Information whether there was a part of a ship in the cell
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/Cell.java]: Better wording for all boolean returns is when it returns true (or false).

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Cell.java [hunk 12]
```diff
@@ -39,24 +39,22 @@ boolean wasShot() {
      * It changes the cell properties.
      *
      * @param newShip
-     * @return
      */
-    Cell addShip(final Ship newShip) {
+    void addShip(Ship newShip) {
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/Cell.java]: All methods which add an element to collection / object come in two flavours. You are switching flavours here. Why? Also, why no longer final?

File: battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java (hunk 1 above)
Comment (rivalez) [reply battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java]: what would you suggest?

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Point2D.java (hunk 5 above)
Comment (rivalez) [reply battleship-client/src/main/java/com/academy/solid/nie/client/ui/Point2D.java]: ok we will change it to instaceof

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java (hunk 10 above)
Comment (rivalez) [reply battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java]: nope, it isn't empty

File: battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java (hunk 4 above)
Comment (rivalez) [reply battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java]: We avoid inline assignment in line 31. So we moved iteration calls to end of the while loop.

File: battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java (hunk 6 above)
Comment (rivalez) [reply battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java]: We are looking for "Provide ships" message which is last of initial messages. All other initial messages are logged. So the answer for your question is: Server should have "Provide ships".

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Cell.java (hunk 12 above)
Comment (rivalez) [reply battleship-client/src/main/java/com/academy/solid/nie/client/ui/Cell.java]: So we decided as a team that we are removing finals in parameters from our code.
Could you explain what do you mean by switching flavours? We don't need this.method to return Cell. We will handle it and rename to setShip, because basically it is just setting Ship to Cell.

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Cell.java (hunk 11 above)
Comment (rivalez) [reply battleship-client/src/main/java/com/academy/solid/nie/client/ui/Cell.java]: ok

File: battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java (hunk 6 above)
Comment (LIttleAncientForestKami) [reply battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java]: I dislike the fact that messages are here treated as a String. If they change, you guys are blindsided to it. Have they been types or enums or enum fields, that would've made you safer and this code - more extendable. Feel free to log this as an issue for later and proceed.

File: battleship-client/src/main/java/com/academy/solid/nie/client/communication/Server.java [hunk 13]
```diff
@@ -13,9 +13,9 @@
  */
 public interface Server {
     /**
-     * Returns true if connection was successful.
-     *
+     * Tries to connect to server.
      * @param ip as String
+     * @throws IOException when there is no server with provided ip
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/communication/Server.java]: Please capitalize names, like IP.

File: battleship-client/src/main/java/com/academy/solid/nie/client/communication/Server.java [hunk 14]
```diff
@@ -35,14 +35,11 @@
 
 
     /**
-     * Receive enemy moves.
-     *
      * @return collection of posiitions
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/communication/Server.java]: Double i in positions.

File: battleship-client/src/main/java/com/academy/solid/nie/client/communication/Server.java [hunk 15]
```diff
@@ -35,14 +35,11 @@
...
      */
     List<Point2D> receiveEnemyMoves();
 
     /**
-     * Receive ships.
      * @return ships positions as string
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/communication/Server.java]: String is a Java type, please write it as expected.

File: battleship-client/src/main/java/com/academy/solid/nie/client/communication/Server.java [hunk 16]
```diff
@@ -35,14 +35,11 @@
...
 
     /**
-     * Receive enemy moves.
-     *
      * @return collection of posiitions
      */
     List<Point2D> receiveEnemyMoves();
 
     /**
-     * Receive ships.
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/communication/Server.java]: :+1:

File: battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java (hunk 4 above)
Comment (LIttleAncientForestKami) [reply battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java]: OK thanks.

File: battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java (hunk 1 above)
Comment (LIttleAncientForestKami) [reply battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java]: Enum. I've briefed Dominik on it.

File: battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java (hunk 6 above)
Comment (LIttleAncientForestKami) [reply battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java]: Thanks for the explanation.

File: battleship-client/src/main/java/com/academy/solid/nie/client/language/Communicate.java [hunk 17]
```diff
@@ -7,5 +7,5 @@
     /**
      * Messages in game.
      */
-    WELCOME, WRONG_IP, WIN, INSERT_IP, CONNECT, LOSE
+    WELCOME, WRONG_IP, WIN, INSERT_IP, CONNECT, RANDOM, LOSE
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/language/Communicate.java]: A good idea. Think about adding "Providing ships" perhaps. ;-)

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Board.java [hunk 18]
```diff
@@ -142,7 +142,8 @@ private void markEndOfShip() {
      * @param ship - represents single instance of Ship
      */
     private boolean canPlaceShip(final Ship ship) {
-        return ship.getPositions().stream().noneMatch(point2D -> !isInScope(point2D) || boardFX.isOccupied(point2D) || isThereShipInTheNeighborhood(point2D));
+        return ship.getPositions().stream().noneMatch(point2D ->
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/Board.java]: The code is pretty and I like the streaming. Problem - you are mixing your game logic here WITH JavaFX code. I'd advise splitting this more decisively, but won't block a PR over it.

I'm not addressing just this line or this method, I'm rather speaking about whole Board class.

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/BoardFX.java [hunk 19]
```diff
@@ -67,11 +67,12 @@ private void createCellInRow(final EventHandler<? super MouseEvent> handler,
      * Indicates the cell as occupied by the ship.
      * The cell color changes to white with a green border.
      *
-     * @param ship - represents single instance of Ship
+     * @param ship  - represents single instance of Ship
      * @param point - represents single Point of the board
      */
     void markFieldAsOccupiedByShip(Ship ship, Point2D point) {
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/BoardFX.java]: Good idea to hide this in a `markField` method.

Comment (LIttleAncientForestKami) [reply battleship-client/src/main/java/com/academy/solid/nie/client/ui/BoardFX.java]: Out of curiosity, I'd like knowing what will use it. Board has BoardFx field. Anything else, that warrants this being package private method?

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Cell.java [hunk 20]
```diff
@@ -35,28 +35,26 @@ boolean wasShot() {
...
         this.ship = newShip;
-        return this;
     }
 
     /**
      * Changes the cell color and marks it as a hit.
      * If there was a piece of a ship in the cell, it would return true.
      *
-     * @return
+     * @return true if there was a part of a ship in the cell
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/Cell.java]: :+1:

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/GameScene.java [hunk 21]
```diff
@@ -35,10 +35,8 @@
     private Parent createContent() {
         BorderPane root = new BorderPane();
         root.setPrefSize(DEFAULT_ROOT_WIDTH, DEFAULT_ROOT_HEIGHT);
-        Button button = new Button();
-        button.setText(CommunicateProviderImpl.getCommunicate(Communicate.WRONG_IP));
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/GameScene.java]: FYI, this is a literal translation of Polish "komunikat", but I don't think English has a noun `communicate` and quick check at Merriam-Webster confirms this. Consider checking few other dictionaries and proving me wrong or changing these class names to Message or Info or Missive or Communique (this one is pretty official).

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/GameScene.java [hunk 22]
```diff
@@ -50,6 +48,13 @@ private Parent createContent() {
         return root;
     }
 
+    private Button createRandomButton() {
+        Button button = new Button();
+        button.setText(CommunicateProviderImpl.getCommunicate(Communicate.RANDOM));
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/GameScene.java]: As above.

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/GameScene.java [hunk 23]
```diff
@@ -72,7 +77,7 @@ private void handlePlayersMove(final Cell cell) {
                     .withButtonWhoExitSystem().display();
             socketServer.sendGameOverToOpponent();
         }
-        if (enemyBoard.isShipSunken(cell.getShip())){
+        if (enemyBoard.isShipSunken(cell.getShip())) {
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/GameScene.java]: What would be needed to make this `ship.isSunken` instead? Feel free to answer me later.

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Point2D.java [hunk 24]
```diff
@@ -23,15 +23,25 @@ public String toString() {
         return String.format("%d,%d,", x, y);
     }
 
+    /**
+     * @param o represent another Point2D
+     * @return information whether two points are the same
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/Point2D.java]: More precise: `true if both points are same or equal`

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Point2D.java [hunk 25]
```diff
@@ -23,15 +23,25 @@ public String toString() {
...
+     */
     @Override
     public boolean equals(Object o) {
-        if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
+        if (this == o) {
+            return true;
+        }
+        if (o == null || !(o instanceof Point2D)) {
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/Point2D.java]: Good call!

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Point2D.java [hunk 26]
```diff
@@ -23,15 +23,25 @@ public String toString() {
...
+            return false;
+        }
         Point2D point2D = (Point2D) o;
-        return x == point2D.x &&
-                y == point2D.y;
+        return x == point2D.x && y == point2D.y;
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/Point2D.java]: :+1:

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Point2D.java [hunk 27]
```diff
@@ -23,15 +23,25 @@ public String toString() {
...
     }
 
+    /**
+     * @return HashCode based of values of fields
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/Point2D.java]: Why is hash code here written as one word with H and C capitalized? :puzzled:

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Ship.java [hunk 28]
```diff
@@ -9,7 +8,7 @@
 class Ship {
     private int health;
 
-    public List<Point2D> getPositions() {
+    List<Point2D> getPositions() {
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/Ship.java]: :+1:

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Ship.java [hunk 29]
```diff
@@ -20,15 +19,15 @@
...
+    void hit() {
         health--;
     }
 
     boolean isAlive() {
         return health > 0;
     }
 
-    int getLength() {
+    private int getLength() {
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/Ship.java]: :+1:

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java [hunk 30]
```diff
@@ -5,26 +5,28 @@
...
+import java.util.List;
+import java.util.Queue;
+import java.util.Random;
 import java.util.stream.IntStream;
 
-
-/**
- *
- */
 class ShipPlacer {
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java]: A ShipPlacerAndMessageToStringConverter would be truer name, would you agree?

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java [hunk 31]
```diff
@@ -5,26 +5,28 @@
...
+    private static final String SHIPS_SEPARATOR = ",\\|";
+    private static final String COORDINATES_SEPARATOR = ",";
     private static final int FOUR_MAST = 4;
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java]: I'd recommend having an Enum for a fleet instead. You can then just iterate over its values and you will know when it's empty fairly easily.

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java [hunk 32]
```diff
@@ -41,21 +43,21 @@
...
             }
         };
     }
 
-    private Ship makeShip(Cell cell, Type type) {
-        return new Ship(producePoints(cell, type));
+    private void finishSetup() {
+        if (typesOfShips.isEmpty()) {
+            socketServer.send(playerBoard.getAllPositions());
+            placeShipsOfEnemy(socketServer.receiveAllShips());
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java]: Minor: `placeEnemyShips` is less of a mouthful.

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java [hunk 33]
```diff
@@ -41,21 +43,21 @@
...
+            areAllShipsPlaced = true;
+        }
     }
 
-    boolean areAllShipsPlaced() {
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java]: :+1:

File: battleship-client/src/main/resources/ENGLISH.properties [hunk 34]
```diff
@@ -3,4 +3,6 @@ WRONG_IP=Wrong IP. Try again!
 WIN= You won!
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/resources/ENGLISH.properties]: Why the spaces on WIN and LOSE?

Comment (LIttleAncientForestKami) [reply battleship-client/src/main/resources/ENGLISH.properties]: Also, you may want to quote that... or ignore me, that may be Bash speaking.

File: battleship-client/src/main/resources/POLISH.properties [hunk 35]
```diff
@@ -3,4 +3,6 @@ WRONG_IP=Nieprawid\u0142owe IP!
 WIN= Wygra\u0142es!
 LOSE= Przegra\u0142es!
 INSERT_IP=Podaj IP:
-CONNECT=Po\u0142\u0105cz!
\ No newline at end of file
+CONNECT=Po\u0142\u0105cz!
+CLEAR=Wyczy\u015B\u0107 plansz\u0119!
+RANDOM=Generuj losowe ustawienie statk\u00F3w!
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/resources/POLISH.properties]: Losuj statki?

File: battleship-client/src/main/resources/POLISH.properties [hunk 36]
```diff
@@ -3,4 +3,6 @@ WRONG_IP=Nieprawid\u0142owe IP!
 WIN= Wygra\u0142es!
 LOSE= Przegra\u0142es!
 INSERT_IP=Podaj IP:
-CONNECT=Po\u0142\u0105cz!
\ No newline at end of file
+CONNECT=Po\u0142\u0105cz!
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/resources/POLISH.properties]: :+1: on UTF.

File: battleship-client/src/test/java/com/academy/solid/nie/client/ui/BoardTest.java [hunk 37]
```diff
@@ -24,7 +23,7 @@ public void shouldPlaceShip_whenThereAreNoShips() {
         EventHandler<MouseEvent> handler = Mockito.mock(EventHandler.class);
         board.initialize(handler);
 
-        new Cell(point).addShip(ship);
+        new Cell(point).setShip(ship);
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/test/java/com/academy/solid/nie/client/ui/BoardTest.java]: All calls like `Assert.assertTrue()` should be made just `assertTrue` through static imports.

File: battleship-client/src/test/java/com/academy/solid/nie/client/ui/BoardTest.java [hunk 38]
```diff
@@ -24,7 +23,7 @@ public void shouldPlaceShip_whenThereAreNoShips() {
...
 
         //when
         boolean result = board.isShipPositionValid(ship);
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/test/java/com/academy/solid/nie/client/ui/BoardTest.java]: Result is a weak name here, it makes your assertion useless. When this fails I'll just learn that result wasn't true. What was the result? I won't know without looking in the code.

File: battleship-client/src/test/java/com/academy/solid/nie/client/ui/BoardTest.java (hunk 37 above)
Comment (LIttleAncientForestKami) [inline battleship-client/src/test/java/com/academy/solid/nie/client/ui/BoardTest.java]: Why do you need this call? It's not used, you're not capturing the reference. Aside from using CPU and RAM this does nothing?

File: battleship-client/src/test/java/com/academy/solid/nie/client/ui/BoardTest.java [hunk 39]
```diff
@@ -43,7 +42,7 @@ public void shouldNotPlaceShip_WhenThereIsAlreadyShip() {
         Ship ship = new Ship(Collections.singletonList(pointA));
 
         Point2D pointB = Point2D.of(0, 0);
-        new Cell(pointB).addShip(ship);
+        new Cell(pointB).setShip(ship);
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/test/java/com/academy/solid/nie/client/ui/BoardTest.java]: As above.

Comment (LIttleAncientForestKami) [reply battleship-client/src/test/java/com/academy/solid/nie/client/ui/BoardTest.java]: And same for other 2.

File: checkstyle.xml [hunk 40]
```diff
@@ -140,7 +140,6 @@
 
         <!-- Checks for common coding problems               -->
         <!-- See http://checkstyle.sf.net/config_coding.html -->
-        <module name="AvoidInlineConditionals"/>
```
Comment (LIttleAncientForestKami) [inline checkstyle.xml]: Why the change?

File: generateReports.sh [hunk 41]
```diff
@@ -8,6 +8,7 @@ function all(){
     echo "Generating all reports. "
     mvn clean
     mvn install site
+	sleep 1
```
Comment (LIttleAncientForestKami) [inline generateReports.sh]: Why the indentation?

File: generateReports.sh [hunk 42]
```diff
@@ -32,14 +33,14 @@ function sonar(){
     -Dsonar.host.url=https://sonarcloud.io \
     -Dsonar.organization=smolamarcin-github \
     -Dsonar.login=a7d3b9bd4e1f813bfb10fcf74162522ed2161273
-    xargs -a reports/sonar.txt firefox -new-tab "$line"
+    xargs -a reports/sonar.txt firefox "$line"
 }
 
 function jacoco(){
     set -e
```
Comment (LIttleAncientForestKami) [inline generateReports.sh]: Have you considered making it a first line after the hash-bang one?

File: pom.xml [hunk 43]
```diff
@@ -204,7 +203,7 @@
                 <version>${mavenCheckstylePluginVersion}</version>
                 <configuration>
                     <configLocation>${checkstyle.config.location}</configLocation>
-                    <failsOnError>true</failsOnError>
```
Comment (LIttleAncientForestKami) [inline pom.xml]: Good call.

File: battleship-client/src/main/resources/POLISH.properties (hunk 35 above)
Comment (EmiliaCiastek) [reply battleship-client/src/main/resources/POLISH.properties]: 'Losuj statki' sounds better

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java [hunk 44]
```diff
@@ -5,26 +5,28 @@
...
-/**
- *
- */
 class ShipPlacer {
+    private static final String SHIPS_SEPARATOR = ",\\|";
+    private static final String COORDINATES_SEPARATOR = ",";
     private static final int FOUR_MAST = 4;
     private static final int THREE_MAST = 3;
     private static final int DOUBLE_MAST = 2;
     private static final int SINGLE_MAST = 1;
```
Comment (EmiliaCiastek) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java]: Enum for ships type would be better

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/BoardFX.java (hunk 19 above)
Comment (Ozymadias) [reply battleship-client/src/main/java/com/academy/solid/nie/client/ui/BoardFX.java]: Only Board use this method

File: battleship-client/src/test/java/com/academy/solid/nie/client/ui/BoardTest.java (hunk 38 above)
Comment (Ozymadias) [reply battleship-client/src/test/java/com/academy/solid/nie/client/ui/BoardTest.java]: should we inline this? or rename to isShipPositionValid?

File: checkstyle.xml (hunk 40 above)
Comment (Ozymadias) [reply checkstyle.xml]: Because we think that ? operator should be allowed

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/BoardFX.java (hunk 19 above)
Comment (LIttleAncientForestKami) [reply battleship-client/src/main/java/com/academy/solid/nie/client/ui/BoardFX.java]: Thank you.

File: battleship-client/src/test/java/com/academy/solid/nie/client/ui/BoardTest.java (hunk 37 above)
Comment (LIttleAncientForestKami) [reply battleship-client/src/test/java/com/academy/solid/nie/client/ui/BoardTest.java]: Bump.

File: battleship-client/src/test/java/com/academy/solid/nie/client/ui/BoardTest.java (hunk 38 above)
Comment (LIttleAncientForestKami) [reply battleship-client/src/test/java/com/academy/solid/nie/client/ui/BoardTest.java]: Inline wins for me. 

```
// when - then

assertTrue(board.isShipPositionValid(ship);
```

File: checkstyle.xml (hunk 40 above)
Comment (LIttleAncientForestKami) [reply checkstyle.xml]: Thank you, I'd agree.
